src/Extend/Settings.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+/*
+ * This file is part of Flarum.
+ *
+ * For detailed copyright and license information, please view the
+ * LICENSE file that was distributed with this source code.
+ */
+
+namespace Flarum\Extend;
+
+use Flarum\Api\Serializer\AbstractSerializer;
+use Flarum\Api\Serializer\ForumSerializer;
+use Flarum\Extension\Extension;
+use Flarum\Foundation\ContainerUtil;
+use Flarum\Settings\SettingsRepositoryInterface;
+use Illuminate\Contracts\Container\Container;
+
+class Settings implements ExtenderInterface
+{
+    private $settings = [];
+    private $defaults = [];
+    private $modifiers = [];
+
+    /**
+     * Serialize a setting value to the ForumSerializer attributes.
+     *
+     * @param string $attributeName: The attribute name to be used in the ForumSerializer attributes array.
+     * @param string $key: The key of the setting.
+     * @return $this
+     */
+    public function serializeToForum(string $attributeName, string $key)
+    {
+        $this->settings[$key] = $attributeName;
+
+        return $this;
+    }
+
+    /**
+     * Set a default value for a setting selected for serialization.
+     *
+     * @param string $key
+     * @param $value
+     * @return $this
+     */
+    public function default(string $key, $value)
```

Comment:
For defaults, I feel like we should somehow modify `DatabaseSettingsRepository` or `MemoryCacheSettingsRepository`, instead of just configuring settings for serialization. Perhaps we could create a `DefaultSettingsManager` class (which could be injected into the settings repos) or even a class wrapping `MemoryCacheSettingsRepository`? That also begs the question: if a setting isn't found in the database and there's an extender-provided default but the user provides a default value in `get`, which should be used? I lean towards treating the extender-provided default as the actual value if there's no actual value regardless of what the user provides.

Reply:
I like that, I was thinking that this would be somewhat misleading if all it does is fallback to a default value when serializing only.

a manager class to inject should be fine, not sure about the naming depending about what we decide in the below comment.

> That also begs the question: if a setting isn't found in the database and there's an extender-provided default but the user provides a default value in get, which should be used?

Logically, it should be
-> database value
-> default value provided in the extender
-> default value provided as a parameter